types/react/index.d.ts

```diff
@@ -2558,6 +2558,14 @@ declare namespace React {
 
     type ValidationMap<T> = PropTypes.ValidationMap<T>;
 
+    type WeakValidationMap<T> = {
```

Comment:
The weak validation map will transform any prop declaration that can be `null` or `undefined` and widen it to be both: `null | undefined`. This allows props authors to declare something like this:
```ts
type Props = { foo?: string }
```

and have it pass checks for HOCs that require it to be:
```ts
type Props = { foo?: string | null }
```

Most developers do not realize that `propTypes` allows for both so this should meet a compromise of DX and runtime accuracy.

Reply:
Hi @ferdaber, after upgrading `@types/react` to 16.7.14 from 16.7.13, my project is getting

```
TS2605: JSX element type 'ReactElement<any> | null' is not a constructor function for JSX elements.
  Type 'ReactElement<any>' is not assignable to type 'Element'.
```

for almost every component. I've tracked down this bug into these changes. I'm wondering if anyone else is getting this error...I'll try to create a reproducible repo soon.

Reply:
Actually, the problem was a different issue (cc @cuyl) - I've written the fix here:

https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17161#issuecomment-446933731

Reply:
thx